### DISCUSSION TITLE
Allow user to configure proxyTimeout value.

### DISCUSSION THREAD

**Discussion by ripvannwinkler** [vercel/next.js#24805] [Ideas]
### Describe the feature you'd like to request

Proxy timeout should be a user-configurable option. 30 seconds is a good default, but the user should be able to adjust if they are aware of the consequences. In my use case, I have a report link that builds an Excel file which can often take 30+ seconds. It's an infrequently accessed API but a critical one nevertheless. Because the proxyTimeout is hard coded in next-server, I'm going to have to resort to using an nginx proxy up front or something. Not ideal.

### Describe the solution you'd like

Expose the proxyTimeout value as a config option.

### Describe alternatives you've considered

There are no alternatives internal to next.js. At present, I'm going to have to add something like nginx to the mix where it was not previously needed.

**Comment by CubedEye** [comment]
We would also like this to be configurable, we're currently using `patch-package` to overwrite it, but it's slowing down the update flow having to repatch this value every update, when it could be a simple configuration valuable

**Reply by lau-dialpad** [reply]
Could you please share where did you patch in order to augment the default time ?
